package/MDAnalysis/analysis/distances.py

```diff
@@ -120,7 +126,8 @@ def contact_matrix(coord, cutoff=15.0, returntype="numpy", box=None):
         return sparse_contacts
 
 
-def dist(A, B, offset=0, box=None):
+def dist(A: AtomGroup, B: AtomGroup, offset: Union[int, Tuple] = 0,
```

Comment:
```suggestion
def dist(A: AtomGroup, B: AtomGroup, offset: Union[int, Tuple[int, int]] = 0,
```

This should remove the need to explicitly convert offset to int later in the function.